pkg/networking/api.go

```diff
@@ -0,0 +1,75 @@
+package networking
```

Comment:
suggestion: I would like to move this file into "internal" since I'm not sure yet how we want to support developers that use this library to access the API. So instead of prematurely committing to a public API (pkg) I would rather use it as an internal util.